### PR TITLE
IBFT 2.0 Doc

### LINKED ISSUE #481: Document how to manage IBFT
### Description
As a user, I want to know how to manage IBFT so that I can use it effectively. 

### Acceptance Criteria
* Documentation written describing how to manage IBFT
* Documentation tested

### PR DESCRIPTION
**Pull request by MadelineMurray**
## Fixed Issue(s)
fixes #481

### CODE REVIEW

File: docs/Consensus-Protocols/IBFT.md [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+using the [`--rpc-enabled`](../Reference/Pantheon-CLI-Syntax.md#rpc-enabled) option. If also using the 
+[`--rpc-api`](../Reference/Pantheon-CLI-Syntax.md#rpc-api) option, include `IBFT`.
+
+The JSON-RPC methods to add or remove validators are:
+
+* [ibft_getPendingVotes](../Reference/JSON-RPC-API-Methods.md#ibft_getPendingVotes)
+* [ibft_proposeValidatorVote](../Reference/JSON-RPC-API-Methods.md#ibft_proposeValidatorVote)
+* [ibft_discardValidatorVote](../Reference/JSON-RPC-API-Methods.md#ibft_discardValidatorVote)
+
+To propose adding a validator, call `ibft_proposeValidatorVote` specifying the address of the proposed signer and `true`.
```
Comment (rain-on) [inline docs/Consensus-Protocols/IBFT.md]: ... of the proposed validator ... (not signer).

maybe "To propose adding a validator, call ... specifying the address of the peer/node to be added, and "true"/

File: docs/Consensus-Protocols/IBFT.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+!!! example "JSON-RPC ibft_proposeValidatorVote Request Example"
+    ```bash
+    curl -X POST --data '{"jsonrpc":"2.0","method":"ibft_proposeValidatorVote","params":["0xFE3B557E8Fb62b89F4916B721be55cEb828dBd73", true], "id":1}' <JSON-RPC-endpoint:port>
+    ``` 
+
+When the next block is created by the validator, a vote is added to the block for the proposed validator.
```
Comment (rain-on) [inline docs/Consensus-Protocols/IBFT.md]: A validator will insert one of the proposals received via JSON RPC into the next block(s) they propose.
If all proposals have been published in this way, subsequent blocks proposed by said validator will not contain a vote.
However, following an epoch (i.e. when all outstanding votes discarded), the locally stored proposals will be re-published.

File: docs/Consensus-Protocols/IBFT.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+
+To propose adding a validator, call `ibft_proposeValidatorVote` specifying the address of the proposed signer and `true`.
+!!! example "JSON-RPC ibft_proposeValidatorVote Request Example"
+    ```bash
+    curl -X POST --data '{"jsonrpc":"2.0","method":"ibft_proposeValidatorVote","params":["0xFE3B557E8Fb62b89F4916B721be55cEb828dBd73", true], "id":1}' <JSON-RPC-endpoint:port>
+    ``` 
+
+When the next block is created by the validator, a vote is added to the block for the proposed validator.  
+
+When more than half of the existing validators propose adding the validators and their votes have been
```
Comment (rain-on) [inline docs/Consensus-Protocols/IBFT.md]: When more than half of the existing validators have published a matching proposal/vote, the proposal is actioned, and the subject of the proposal will be added/removed from the validator pool accordingly.

File: docs/Consensus-Protocols/IBFT.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    ```bash
+    curl -X POST --data '{"jsonrpc":"2.0","method":"ibft_discardValidatorVote","params":["0xFE3B557E8Fb62b89F4916B721be55cEb828dBd73"], "id":1}' <JSON-RPC-endpoint:port>
+    ```
+
+The process for removing a validator is the same as adding a validator except you specify `false` as the second parameter of `ibft_proposeValidatorVote`. 
+
+### Epoch Transition
+
+At each epoch transition, all pending votes collected from received blocks are discarded. Existing proposals remain 
+in effect and validators re-add their vote the next time they create a block.
```
Comment (rain-on) [inline docs/Consensus-Protocols/IBFT.md]: given this is here - you can probably discard half of my previous comment.

File: docs/Consensus-Protocols/IBFT.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+!!! example "JSON-RPC ibft_discardValidatorVote Request Example"
+    ```bash
+    curl -X POST --data '{"jsonrpc":"2.0","method":"ibft_discardValidatorVote","params":["0xFE3B557E8Fb62b89F4916B721be55cEb828dBd73"], "id":1}' <JSON-RPC-endpoint:port>
+    ```
+
+The process for removing a validator is the same as adding a validator except you specify `false` as the second parameter of `ibft_proposeValidatorVote`. 
+
+### Epoch Transition
+
+At each epoch transition, all pending votes collected from received blocks are discarded. Existing proposals remain
```
Comment (rain-on) [inline docs/Consensus-Protocols/IBFT.md]: Insert before this something like (maybe) (ok, this is probably just a reording of what is here).

An 'Epoch Transition' occurs every X blocks - where X is defined in the Ibft genesis file (epochBlockLength?).

File: docs/Reference/Pantheon-CLI-Syntax.md [hunk 6]
```diff
@@ -285,7 +285,7 @@ You must specify a valid coinbase when you enable mining using the [`--miner-ena
 option or the [`miner_start`](JSON-RPC-API-Methods.md#miner_start) JSON RPC-API method.
 
 !!!note
-    This option is ignored in networks using the [Clique Proof-of-Authority (PoA) consensus protocol](../Configuring-Pantheon/Proof-of-Authority.md). 
+    This option is ignored in networks using the [Clique Proof-of-Authority (PoA) consensus protocol](../Consensus-Protocols/Clique.md).
```
Comment (rain-on) [inline docs/Reference/Pantheon-CLI-Syntax.md]: will also be ignored for IBFT 2.0.

File: docs/Consensus-Protocols/IBFT.md (hunk 1 above)
Comment (MadelineMurray) [reply docs/Consensus-Protocols/IBFT.md]: Done

File: docs/Consensus-Protocols/IBFT.md (hunk 2 above)
Comment (MadelineMurray) [reply docs/Consensus-Protocols/IBFT.md]: First two sentences included. Last sentence appears further down.

File: docs/Consensus-Protocols/IBFT.md (hunk 3 above)
Comment (MadelineMurray) [reply docs/Consensus-Protocols/IBFT.md]: Done. (removing is further down the page so have left that out)

File: docs/Consensus-Protocols/IBFT.md (hunk 4 above)
Comment (MadelineMurray) [reply docs/Consensus-Protocols/IBFT.md]: Done

File: docs/Consensus-Protocols/IBFT.md (hunk 5 above)
Comment (MadelineMurray) [reply docs/Consensus-Protocols/IBFT.md]: Done

File: docs/Reference/Pantheon-CLI-Syntax.md (hunk 6 above)
Comment (MadelineMurray) [reply docs/Reference/Pantheon-CLI-Syntax.md]: Done

File: docs/Consensus-Protocols/IBFT.md [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+    ```
+    
+Properties specific to IBFT 2.0 are:
+
+* `blockperiodseconds` - Minimum block time in seconds. 
+* `epochlength` - Number of blocks after which to reset all votes.
+* `requesttimeoutseconds` - Timeout for each consensus round before a round change. 
+* `extraData` - `RLP([32 Bytes Vanity, List<Validators>, No Votes, Round=Int(0), 0 Seals])`
+
+The `extraData` property is RLP (Recursive Length Prefix) encoded. RLP encoding is a space efficient object
```
Comment (NicolasMassart) [inline docs/Consensus-Protocols/IBFT.md]: Example of usage of the abbreviation extension
However, it's better to group all abbreviation definitions at the end of the page.
```suggestion
The `extraData` property is RLP encoded. RLP encoding is a space efficient object 
*[RLP]: Recursive Length Prefix
```

File: docs/Consensus-Protocols/IBFT.md [hunk 8]
```diff
@@ -0,0 +1,117 @@
...
+
+!!! example "Decoding Extra Data Example"
+    
+    Using the [EthereumJS RLP](https://github.com/ethereumjs/rlp) library: 
+    ```bash
+    rlp decode "0xf853a00000000000000000000000000000000000000000000000000000000000000000ea94be068f726a13c8d46c44be6ce9d275600e1735a4945ff6f4b66a46a2b2310a6f3a93aaddc0d9a1c193808400000000c0"
+    ```
+    
+    The decoded result is: 
+    ```
```
Comment (NicolasMassart) [inline docs/Consensus-Protocols/IBFT.md]:
```suggestion
    ```json
```

File: docs/Consensus-Protocols/IBFT.md [hunk 9]
```diff
@@ -0,0 +1,117 @@
...
+
+* `nonce` - `0x0`
+* `difficulty` - `0x1`
+* `mixHash` - `0x63746963616c2062797a616e74696e65206661756c7420746f6c6572616e6365` for Istanbul block identification.
+
+To start a node on an IBFT 2.0 private network, use the [`--genesis`](../Reference/Pantheon-CLI-Syntax.md#genesis`) option to specify the custom genesis file. 
+
+## Adding and Removing Validators
+
+To propose adding or removing validators using the JSON-RPC methods, enable the RPC interface
```
Comment (NicolasMassart) [inline docs/Consensus-Protocols/IBFT.md]: As in "Adding and Removing Signers", I propose to add the WS options too, but it's only if we did it on both commands.

File: docs/Consensus-Protocols/Clique.md [hunk 10]
```diff
@@ -37,9 +37,9 @@ The properties specific to Clique are:
 * `epoch` - Number of blocks after which to reset all votes.
 * `extraData` - Initial signers are specified after the 32 bytes reserved for vanity data.
```
[the remark this answers is not in the document]
Comment (MadelineMurray) [reply docs/Consensus-Protocols/Clique.md]: Done

File: docs/Consensus-Protocols/Clique.md [hunk 11]
```diff
@@ -37,9 +37,9 @@ The properties specific to Clique are:
 * `epoch` - Number of blocks after which to reset all votes.
 * `extraData` - Initial signers are specified after the 32 bytes reserved for vanity data. 
 
-To connect to the Rinkeby testnet, start Pantheon with the [`--rinkeby`](../Reference/Pantheon-CLI-Syntax.md#rinkeby) command line option. To start a node on a PoA private network, use the [`--network-id`](../Reference/Pantheon-CLI-Syntax.md#network-id) command line option and [`--genesis`](../Reference/Pantheon-CLI-Syntax.md#genesis`) option. 
+To connect to the Rinkeby testnet, start Pantheon with the [`--rinkeby`](../Reference/Pantheon-CLI-Syntax.md#rinkeby) command line option. To start a node on a Clique private network, use the [`--genesis`](../Reference/Pantheon-CLI-Syntax.md#genesis`) option to specify the custom genesis file. 
 
-### Adding and Removing Signers
+## Adding and Removing Signers
 
 To propose adding or removing signers using the JSON-RPC methods, you must enable the RPC interface using the [`--rpc-enabled`](../Reference/Pantheon-CLI-Syntax.md#rpc-enabled) option. If also using the [`--rpc-api`](../Reference/Pantheon-CLI-Syntax.md#rpc-api) option, include `CLIQUE`.
```
[the remark this answers is not in the document]
Comment (MadelineMurray) [reply docs/Consensus-Protocols/Clique.md]: Done

File: docs/Consensus-Protocols/IBFT.md (hunk 7 above)
Comment (MadelineMurray) [reply docs/Consensus-Protocols/IBFT.md]: Done

File: docs/Consensus-Protocols/IBFT.md (hunk 9 above)
Comment (MadelineMurray) [reply docs/Consensus-Protocols/IBFT.md]: Done

**Pull request merged**